spp_idqueue/models/id_batch.py

```diff
@@ -25,3 +25,22 @@ class OpenSPPPrintBatch(models.Model):
 
     id_pdf = fields.Binary("ID PASS")
     id_pdf_filename = fields.Char("ID File Name")
+
+    def generate_batch(self):
+        for rec in self:
+            for queue in rec.queued_ids:
+                queue.on_generate()
+            if not rec.queued_ids.filtered(lambda x: x.status not in ["generated"]):
+                rec.status = "generated"
+                rec.pass_api_param()
```

Comment:
You should just call the method `rec.queued_ids.generate_cards()`

Reply:
Done